NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+         day = day(Date)) %>%
+  ungroup()
+
+data_admin3 <- 
+  ndviDataLong3 %>%
+  mutate(year = year(Date), 
+         month = month(Date), 
+         day = day(Date)) %>%
+  ungroup()
+
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Generate Total NDVI Graphics By Admin 0 (country) and Admin 1 (building off admin 2) ---- 
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Admin 0 (one line for the entire region)
+peakndvi <- 
+  yearData_admin2 %>%
+  group_by(year) %>%
+  mutate(meanAnnualndvi = mean(peak_ndvi)) %>%
+  ungroup() %>%
+  distinct(year, meanAnnualndvi)
+
+peakndvi %>% 
+  ggplot(aes(x=year, y=meanAnnualndvi)) +
+  geom_line() + 
+  labs(title="Peak NDVI by Year", x="Year", y ="Peak NDVI") +
+  theme_classic() 
+
+# Calculate total admin 2 and 3 ndvi for maps (if every obs is on admin 2 basis, then show)
+yearData_admin2 <- 
+  data_admin2 %>% 
+  group_by(admin2Name, admin2Pcod, year) %>%
+  mutate(peak_ndvi = max(newNDVI, na.rm = TRUE),
+         mean_ndvi = mean(peak_ndvi, na.rm = TRUE),
+         standardDeviation_ndvi = sd(peak_ndvi, na.rm = TRUE)) %>%
+  distinct(admin1Name, admin1Pcod,admin2Name, admin2Pcod, year, 
+           peak_ndvi, mean_ndvi, standardDeviation_ndvi) %>%
+  ungroup()
+
+yearData_admin3 <- 
+  data_admin3 %>% 
+  group_by(adm_03, rowcacode3, year) %>%
+  mutate(peak_ndvi = max(newNDVI, na.rm = TRUE),
+         mean_ndvi = mean(peak_ndvi, na.rm = TRUE),
+         standardDeviation_ndvi = sd(peak_ndvi, na.rm = TRUE)) %>%
+  distinct(adm_03, rowcacode3, year, 
+           peak_ndvi, mean_ndvi, standardDeviation_ndvi) %>%
+  ungroup()
+# Take the mean over the spatial area for a given year ----
+yearData_admin1 <-
+  yearData_admin2 %>% 
+  group_by(admin1Name, year) %>% 
+  mutate(peak_ndvi_annual_admin1 = mean(peak_ndvi)) %>% 
+  ungroup() %>% 
+  distinct(admin1Name, peak_ndvi_annual_admin1, year)
+
+### Generate Overall ndvi  Graphic ---- 
+yearData_admin1  %>% 
+  filter(year < 2022) %>% 
+  ggplot(aes(x = year, 
+             y = peak_ndvi_annual_admin1, 
+             color = admin1Name, linetype = admin1Name)) +
+  geom_line() +
+  labs(title = "Peak NDVI By Administrative Region (Admin 1)", 
+       caption = "Data Source: NOAA CDR AVHRR NDVI",
+       y = "", 
+       x = "",
+       color = "Region",
+       linetype = "Region") +
+  remove_chart_clutter +
+  theme(legend.position = "right")
+
+# What's the avg peak ndvi per year in Niger?
+mean(yearData_admin1$peak_ndvi_annual_admin1, na.rm = TRUE)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Generate Peak NDVI Maps (Admin 2) -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+# 2. Merge Precip and  Spatial Information (geometry) by common code
+nigerShpMerged_admin2 = full_join(nigerMapAdmin2, 
+                                  yearData_admin2,
+                                  by=c("admin2Pcod" = "admin2Pcod"))
```

Comment:
here you can see why it can be frustrating to have slight differences in cases -- consistently using lowercase helps avoid this problem. in the janitor package, using ` clean_names()` after you read in your dataset will automatically convert your columns into a tidyverse compatible, lowercase syntax.